pages/_app/index.jsx

```diff
@@ -78,41 +76,70 @@ class App extends NextApp {
     }
   }
 
-  fetchUser = async () => {
-    try {
-      const { dataProvider, activity } = new Route(window.location.pathname)
-      await this.store.init(dataProvider, activity)
-      this.setState({ isAuthorized: true })
-      Sentry.configureScope((scope) => {
-        scope.setUser({
-          id: this.store.user.id,
-          email: this.store.user.email,
-        })
-      })
-    } catch (unauthorizedError) {
-      // TODO: Do some check before redirect
-      this.setState({ isAuthorized: false })
-      if (!this.isRouteWithoutStore) this.redirectToLogin()
-      throw unauthorizedError
-    }
+  get isAuthorized() {
+    return !!this.store?.user?.id
   }
 
-  get isRouteWithoutStore() {
-    const { router } = this.props
-    return ROUTES_WITHOUT_STORE.find((r) => router.asPath.startsWith(r))
-  }
+  componentDidMount() {
+    window.addEventListener('unhandledrejection', this.handlePromiseRejection)
 
-  async componentDidMount() {
     const { router } = this.props
+    router.events.on('routeChangeComplete', logPageView)
 
-    window.addEventListener('unhandledrejection', this.handlePromiseRejection)
+    if (isRouteWithoutStore(router.asPath)) return
+
+    const { store } = this
+    const initializeUser = store.init()
+
+    /**
+     * TODO: Remove the following behaviour, find another option to handle this:
```

Comment:
I would rather suggest to not merging this. I mean I would prefer to push only the part with activities. I know it took you so much time but if we push this "dangerous code" it won't be touch. It will stay there forever. 😞 It's very similar to [this](https://github.com/oacore/dashboard/blob/master/pages/data-providers/%5Bdata-provider-id%5D/deposit-compliance.jsx#L140) 


I am not going to force you. It's your call here. 🙂

Reply:
Honestly, I am not sure about the comment you reference to, it seems to be okay with me now. Also, this comment is going to be resolved in some way with templates like in #305, even if not, it should be removed.

About this, I wanted to merge this and refactor App straight after this PR. However, I am not sure we have a time frame for this refactoring so I will consider removing it since you have a strong point.

Thank you for the feedback! 🙂

Reply:
I checked it one more time trying to parse the route during the initialisation to avoid dangerous code. I am afraid it's not possible and we have no other option at the moment.

The initial router state looks like this:
```js
{
  asPath: "/data-providers/[data-provider-id]/overview",
  pathname: "/data-providers/[data-provider-id]/overview",
  route: "/data-providers/[data-provider-id]/overview",
  query: {}
}
```

Not only `query` is not populated, but all parameters are not parsed.

Reply:
there is an option. parse `window.location` directly.

Reply:
🤦‍♂️

Reply:
Does the current option satisfy you?

Reply:
yep, thank you! 🙂